### ISSUE TITLE
Error on sending message to Facebook status Forbidden

### ISSUE THREAD

**Issue by JamesLouie** [Microsoft/BotBuilder#1246]
I have been managing a bot for a couple months now, and I recently ran into an issue where Facebook is having intermittent issues connecting with the Bot Connector.  While in dialog with Facebook, we're seeing that at random, some messages get through to the Facebook chat client while others return as 500s.  This ranges from sending a generic typing message to responses generated from a FormFlow dialog.

The exception raised:
```
System.UnauthorizedAccessException: Authorization for Microsoft App ID 4b6154c8-3228-4403-8999-dc7ae63c7d5c failed with status code Forbidden
   at Microsoft.Bot.Connector.JwtTokenRefresher.<SendAsync>d__2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.Bot.Connector.Conversations.<ReplyToActivityWithHttpMessagesAsync>d__7.MoveNext()
```

Our slack channel is still functioning as expected.  

I did a run through of #1105 which included:
- Checking my App Credentials
- Checking the BotAuthentication annotation
- Checking the trusted service url
- Use Bot Framework portal to test connection

My current project setup is:
- C#
- Bot Builder SDK 3.2.1
- Channels: Slack and Facebook

One of our calls that generate this error:
```cs
			Activity reply = activity.CreateReply();
			reply.Type = ActivityTypes.Typing;
			reply.Text = null;
			await connector.Conversations.ReplyToActivityAsync(reply);
```

**Comment by wiltodelta** [comment]
Confirm problem with Facebook.
<img width="754" alt="screen shot 2016-09-16 at 08 30 07" src="https://cloud.githubusercontent.com/assets/1192743/18576194/fc3d2fbe-7be7-11e6-8b03-23d49fdb9b9d.png">

**Comment by kenyeung128** [comment]
it's the same for me as well

**Comment by kenyeung128** [comment]
i can reproduce the same issues with EchoBot from the sample project of BotBuilder, some messages get through but some messages return 500 with UnauthorizedAccessException. Is there any workaround for now? as it's just stop the bot working.

**Comment by alonsonic** [comment]
Having the same issue as wiltodelta

**Comment by sMorac** [comment]
I'm seeing this on Node.js as well, some messages go through and some are 403. After testing it on Facebook and Skype, only the Facebook channel has the problem.

**Comment by gobaldia** [comment]
@wiltodelta there's an open issue #1254 
Please comment something there to show Microsoft's team that's a general problem.

**Comment by dandriscoll** [comment]
Hi folks, we're looking into it. Thanks

**Comment by JamesLouie** [comment]
Verifying on my side that I think this issue has been resolved along with #1254 and #1257.

Thank you for the quick response!
